File: CS/RemoveGridLinesExample/Form1.cs
using DevExpress.XtraPivotGrid;
using DevExpress.XtraPivotGrid.Localization;
using System;
using System.Data;
using System.Drawing;

namespace RemoveGridLinesExample
{
    public partial class Form1 : DevExpress.XtraEditors.XtraForm
    {
        public Form1()
        {
            PivotGridLocalizer.Active = new MyLocalizer();
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            pivotGridControl1.BeginUpdate();
            pivotGridControl1.DataSource = CreateTable(20);
            InitializeFields();
            pivotGridControl1.CustomDrawFieldValue += pivotGridControl1_CustomDrawFieldValue;
            pivotGridControl1.Appearance.Lines.BackColor = Color.Transparent;
            pivotGridControl1.EndUpdate();

        }

        void pivotGridControl1_CustomDrawFieldValue(object sender, PivotCustomDrawFieldValueEventArgs e)
        {
            if (e.Area == PivotArea.ColumnArea)
            {
                e.Graphics.FillRectangle(Brushes.White, e.Bounds);
                e.Handled = true;
            }

        }

        private void InitializeFields()
        {
            pivotGridControl1.Fields.Add("Type", PivotArea.RowArea);
            pivotGridControl1.Fields.Add("Product", PivotArea.RowArea);
            PivotGridField fieldYear = new PivotGridField("Date", PivotArea.RowArea);
            fieldYear.Name = "FieldYear";
            fieldYear.Caption = fieldYear.Name;
            fieldYear.GroupInterval = PivotGroupInterval.DateYear;
            PivotGridField fieldMonth = new PivotGridField("Date", PivotArea.RowArea);
            fieldMonth.Name = "FieldMonth";
            fieldMonth.Caption = fieldMonth.Name;
            fieldMonth.GroupInterval = PivotGroupInterval.DateMonth;
            pivotGridControl1.Fields.AddRange(new PivotGridField[] { fieldYear, fieldMonth });
            pivotGridControl1.Fields.Add("Flag", PivotArea.RowArea);
        }
        private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Type", typeof(string));
            tbl.Columns.Add("Product", typeof(string));
            tbl.Columns.Add("Date", typeof(DateTime));
            tbl.Columns.Add("Flag", typeof(bool));

            for (int i = 0; i < RowCount; i++)

                for (int j = 0; j < RowCount; j++)
                    tbl.Rows.Add(new object[] {
                        String.Format("Type {0}", i % 5),
                        String.Format("Product {0}", i % 3),
                        DateTime.Now.AddYears(j % 7).AddMonths(j % 9), i % 2 == 0 });
            return tbl;
        }

    }
    public class MyLocalizer : PivotGridResLocalizer
    {
        public override string GetLocalizedString(PivotGridStringId id)
        {
            if (id == PivotGridStringId.DataHeadersCustomization
                || id == PivotGridStringId.ColumnHeadersCustomization
                || id == PivotGridStringId.FilterHeadersCustomization)
                return String.Empty;
            return base.GetLocalizedString(id);
        }
    }

}

File: VB/RemoveGridLinesExample/Form1.vb
Imports DevExpress.XtraPivotGrid
Imports DevExpress.XtraPivotGrid.Localization
Imports System
Imports System.Data
Imports System.Drawing

Namespace RemoveGridLinesExample
	Partial Public Class Form1
		Inherits DevExpress.XtraEditors.XtraForm

		Public Sub New()
			PivotGridLocalizer.Active = New MyLocalizer()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			pivotGridControl1.BeginUpdate()
			pivotGridControl1.DataSource = CreateTable(20)
			InitializeFields()
			AddHandler pivotGridControl1.CustomDrawFieldValue, AddressOf pivotGridControl1_CustomDrawFieldValue
			pivotGridControl1.Appearance.Lines.BackColor = Color.Transparent
			pivotGridControl1.EndUpdate()

		End Sub

		Private Sub pivotGridControl1_CustomDrawFieldValue(ByVal sender As Object, ByVal e As PivotCustomDrawFieldValueEventArgs)
			If e.Area = PivotArea.ColumnArea Then
				e.Graphics.FillRectangle(Brushes.White, e.Bounds)
				e.Handled = True
			End If

		End Sub

		Private Sub InitializeFields()
			pivotGridControl1.Fields.AddDataSourceColumn("Type", PivotArea.RowArea)
			pivotGridControl1.Fields.AddDataSourceColumn("Product", PivotArea.RowArea)
			Dim fieldYear As New PivotGridField()
			fieldYear.Area = PivotArea.RowArea
			fieldYear.DataBinding = New DataSourceColumnBinding("Date", PivotGroupInterval.DateYear)
			fieldYear.Name = "FieldYear"
			fieldYear.Caption = fieldYear.Name
			Dim fieldMonth As New PivotGridField()
			fieldMonth.Area = PivotArea.RowArea
			fieldMonth.DataBinding = New DataSourceColumnBinding("Date", PivotGroupInterval.DateMonth)
			fieldMonth.Name = "FieldMonth"
			fieldMonth.Caption = fieldMonth.Name
			pivotGridControl1.Fields.AddDataSourceColumn("Flag", PivotArea.RowArea)
		End Sub
		Private Function CreateTable(ByVal RowCount As Integer) As DataTable
			Dim tbl As New DataTable()
			tbl.Columns.Add("Type", GetType(String))
			tbl.Columns.Add("Product", GetType(String))
			tbl.Columns.Add("Date", GetType(Date))
			tbl.Columns.Add("Flag", GetType(Boolean))

			For i As Integer = 0 To RowCount - 1

				For j As Integer = 0 To RowCount - 1
					tbl.Rows.Add(New Object() { String.Format("Type {0}", i Mod 5), String.Format("Product {0}", i Mod 3), Date.Now.AddYears(j Mod 7).AddMonths(j Mod 9), i Mod 2 = 0 })
				Next j
			Next i
			Return tbl
		End Function

	End Class
	Public Class MyLocalizer
		Inherits PivotGridResLocalizer

		Public Overrides Function GetLocalizedString(ByVal id As PivotGridStringId) As String
			If id = PivotGridStringId.DataHeadersCustomization OrElse id = PivotGridStringId.ColumnHeadersCustomization OrElse id = PivotGridStringId.FilterHeadersCustomization Then
				Return String.Empty
			End If
			Return MyBase.GetLocalizedString(id)
		End Function
	End Class

End Namespace